webapp/src/snippetBuilder.tsx

```diff
@@ -101,59 +134,86 @@ export class CreateSnippetBuilder extends data.Component<ISettingsProps, CreateS
         });
     }
 
+    /**
+     * Closes the modal
+     */
     cancel() {
         pxt.tickEvent("snippetBuilder.cancel", undefined, { interactiveConsent: true });
         this.hide();
     }
 
+    /**
+     * Takes the output from state, runs replace tokens, decompiles the resulting typescript
+     * and outputs the result as a Blockly xmlDOM. This then uses appendDomToWorkspace to attach 
+     * our xmlDOM to the mainWorkspaces passed to the component.
+     */
     injectBlocksToWorkspace() {
-        const { mainWorkspace, output } = this.state;
+        const { mainWorkspace, tsOutput } = this.state;
 
         compiler.getBlocksAsync()
-            .then(blocksInfo => compiler.decompileBlocksSnippetAsync(this.replaceTokens(output), blocksInfo))
+            .then(blocksInfo => compiler.decompileBlocksSnippetAsync(this.replaceTokens(tsOutput), blocksInfo))
             .then(resp => {
-                Blockly.Xml.appendDomToWorkspace(Blockly.Xml.textToDom(resp), mainWorkspace);
+                const xmlDOM = Blockly.Xml.textToDom(resp)
+                Blockly.Xml.appendDomToWorkspace(xmlDOM, mainWorkspace);
             });
     }
 
+    /**
+     * Hides modal and injects our blocks to the work space.
+     */
     confirm() {
         this.injectBlocksToWorkspace();
         this.hide();
     }
 
+    /** 
+     * @param increment - this adds either 1 or -1 to the value of currentQuestion
```

Comment:
like below, this is too prescriptive. "currentQuestion" is an implementation detail that the reader doesn't need to know about. Omitting implementation specifics helps the reader read faster and it makes it easier for the maintainer since they can change things like how "currentQuestion" works without having to come back and change the comments.
Writing comments is a balancing act between being descriptive enough to be useful while still being vague enough to not be locked to one implementation

Reply:
I deleted some of the comments and hopefully am now only describing what could be unclear by reading the code.